openshift/console
# Add getting started section to the add page in developer perspective

**Fixes**: 
Epic: https://issues.redhat.com/browse/ODC-5013
Story: https://issues.redhat.com/browse/ODC-5560

**Analysis / Root cause**: 
As a user, I want to see the new getting started section with areas to "Set up your cluster", Quick Starts, Guided Tours and more Learning resources.

**Solution Description**: 
This PR adds a new section to "+Add" page in the developer perspective above all options to import different types of resources. (This grid of add actions are redesigned as part of issue #8867.)

This new "Getting started" section adds three content cards with different links:

1. "Create applications using samples" - The first column is based on the developer catalog and shows two samples. The code contains two hardcoded "featured" samples (Quarkus and Sprint Boot) which should be preferred. When this samples are not found, some other samples will be displayed.
2. "Build with guided documentation" - This column shows up to two quick starts. It also prefers two hardcoded quickstarts (Quarkus and Sprint Boot again). It but will automatically hide them when the user completed them. When all quick started are completed, this column disappears.
3. "Explore new developer features" shows some static links. Currently it shows a link to `https://developers.redhat.com/products/openshift/getting-started` and two options to open the topology. The first link opens also opens the "Add to project" search bar.

It contains an option to hide this getting started section.

*Note:* The button too restore this section is not correct aligned, as this need some changes from #8867. Depending on which PR is merged first, this PR here or #8867 need to address this.

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux

![image](https://user-images.githubusercontent.com/139310/118571653-d0b1af00-b77e-11eb-84ed-d55ff2f94a06.png)

![Peek 2021-05-18 02-26](https://user-images.githubusercontent.com/139310/118572347-7fa2ba80-b780-11eb-9d62-4a55a326ac96.gif)

**Unit test coverage report**: 
![image](https://user-images.githubusercontent.com/139310/118572185-25095e80-b780-11eb-8f57-36f371627ff0.png)

![image](https://user-images.githubusercontent.com/139310/118572212-32bee400-b780-11eb-9e05-2b2ef7ca6294.png)

**Test setup:**
* Open the developer perspective, Add Page.
* Check the different options.
* Complete a shown quick start and check that another quick start is shown.

**Browser conformance**:
